src/Node/Stream.purs

```diff
@@ -61,6 +66,16 @@ type Writable r = Stream (write :: Write | r)
 -- | A duplex (readable _and_ writable stream)
 type Duplex = Stream (read :: Read, write :: Write)
 
+type Data = Either String Buffer
+
+-- | Read a chunk of data into `Data`
+-- | This is a mere internal plumbing function used by functions reading data
+-- | from a stream.
+type ChunkReader = Foreign -> Data
+readChunk :: ChunkReader
```

Comment:
All these ADTs and type synonyms are just to safe me typing and thinking (i.e. was the `String` `Left` or `Right`?).